packages/comms/src/ecl/dfuWorkunit.ts

```diff
@@ -0,0 +1,332 @@
+import { Cache, IEvent, scopedLogger, StateCallback, StateEvents, StateObject, StatePropCallback } from "@hpcc-js/util";
+import { IConnection, IOptions } from "../connection";
+import { ESPExceptions } from "../espConnection";
+import { WsSMC } from "../services/wsSMC";
+import { FileSpray, SprayFixedEx, FileSprayService, SprayVariableEx } from "../services/fileSpray";
+import * as WsTopology from "../services/wsTopology";
+
+const logger = scopedLogger("dfuWorkunit.ts");
+
+enum States {
+    Unknown = 0,
+    Scheduled,
+    Queued,
+    Started,
+    Aborted,
+    Failed,
+    Finished,
+    Monitoring,
+    Aborting,
+    NotFound = 999
+}
+
+export class DFUWorkunitCache extends Cache<{ BaseUrl: string, ID: string }, DFUWorkunit> {
+    constructor() {
+        super((obj) => {
+            return `${obj.BaseUrl}-${obj.ID}`;
+        });
+    }
+}
+const _workunits = new DFUWorkunitCache();
+
+export type DFUWorkunitEvents = "finished" | StateEvents;
+export type UDFUWorkunitState = FileSpray.DFUWorkunit;
+export type IDFUWorkunitState = FileSpray.DFUWorkunit | WsSMC.ActiveWorkunit;
```

Comment:
For the running DFU Workunits - do they come back as an ActiveWorkunit or a DFUJob?

Reply:
They seem to be ActiveWorkunits; they are mixed in with WUs in the ActiveWorkunit array included in the Activities response.